resourcematch-api/main.py

```diff
@@ -140,21 +156,21 @@ def search_responses_to_urls(search_responses):
         book_data = BOOKS_BY_SLUG[response['book']]
         book_uuid = book_data['uuid']
         search_query = response['search_query']
-        response_hits = response['search_response']['hits']['hits']
-        for hit in response_hits:
-            search_score = hit['_score']
-            element_uuid = hit['_source']['element_id']
-            page_uuid = hit['_source']['page_id']
-            page_uuid_partition = page_uuid.partition('@')[0]
-            page_data = get_book_page_data(book_uuid, page_uuid_partition)
-            page_url_main = page_data['urls']['main']
-            appended_url_element_id = f"{page_url_main}#{element_uuid}"
-            hit_data = {
-                "hit_score": search_score,
-                "hit_query": search_query,
-                "hit_url": appended_url_element_id
-            }
-            unsorted_hits.append(hit_data)
+        response_hit = response['search_response']
+
+        search_score = response_hit['_score']
+        element_uuid = response_hit['_source']['element_id']
+        page_uuid = response_hit['_source']['page_id']
+        page_uuid_partition = page_uuid.partition('@')[0]
+        page_data = get_book_page_data(book_uuid, page_uuid_partition)
```

Comment:
This is more of a note than something I think we need to do anything about for the prototype: We use async to create concurrent search API requests, but we serialize requests for page metadata. I suspect in practice there aren't a ton of unique pages in a search response so caching would minimize spurious requests.

Reply:
Correct - just eyeballing the `page_uuid`s and it seemed like many were the same. If we move forward with this past the prototype then it is something we should account for via caching.